src/app/dataset-view/dataset.service.ts

```diff
@@ -51,10 +49,10 @@ export class AppDatasetService {
         this.datasetChanges$.next(searchDatasetInfo);
     }
 
-    public requestDatasetDataSchema(info: DatasetInfo): void {
+    public requestDatasetMainData(info: DatasetInfo): void {
```

Comment:
Please split this large method into parts. Since now you have named fragments for every part, we can use them to have good types for parameters.